qp_deblur/deblur.py

```diff
@@ -562,31 +562,34 @@ def generate_tree_from_fragments(fp_placements,
     with open(fp_placements) as placements_file:
         placements = json.load(placements_file)
 
-        fp_phylogeny = generate_insertion_trees(
+        try:
+            fp_phylogeny = generate_insertion_trees(
                                     placements,
                                     out_dir,
                                     reference_template=fp_reference_template,
                                     reference_rename=fp_reference_rename)
-
-        # trim BIOM file to match newly-generated tree
-
-        # read tree
-        tree = TreeNode.read(str(fp_phylogeny))
-        fragments_tree = {str(tip.name) for tip in tree.tips()
-                          if tip.name is not None}
-
-        # load biom w/fragments not found in fp_phylogeny
-        biom_table = load_table(fp_biom)
-        fragments_table = set(map(str, biom_table.ids(axis='observation')))
-
-        # filter biom file
-        tbl_matched = biom_table.filter(fragments_table & fragments_tree,
-                                        axis='observation',
-                                        inplace=False)
-
-        # overwrite biom w/pared final biom
-        with biom_open(fp_biom, 'w') as f:
-            tbl_matched.to_hdf5(
-                f, "Generated by Qiita, qp-deblur")
-
-        return fp_phylogeny, fp_biom
+        except Exception:
```

Comment:
It might be good to explicitly state why we're expecting a possible Exception (in this case an issue w/reference_template), but it's not a show-stopper.